scripts/west_commands/runners/core.py

```diff
@@ -153,8 +158,9 @@ def _parse(self, filename: str):
 
     @staticmethod
     def _parse_value(value):
+        # If it is a quoted string, return it with quotation marks removed
         if value.startswith('"') or value.startswith("'"):
-            return value.split()
+            return value[1:-1].strip()
```

Comment:
I think it needs to be stripped first:

```
value.strip()[1:-1]
```

Otherwise, a manually edited file with whitespace at the end of a string option will fail to parse as intended:

```
CONFIG_MY_STRING="foo"        
                      ^^^^
                      manual whitespace
```

Reply:
> 
> 
> I think it needs to be stripped first:
> 
> ```
> value.strip()[1:-1]
> ```
> 
> Otherwise, a manually edited file with whitespace at the end of a string option will fail to parse as intended:
> 
> ```
> CONFIG_MY_STRING="foo"        
>                       ^^^^
>                       manual whitespace
> ```

Hm... I want to keep my `strip`, lets to:
```
value.strip()[1:-1].strip()
```
Every white-space gets stripped.

Reply:
Can you explain why you want to strip whitespace that the user included explicitly?

Reply:
Good point. Somehow I thought that leading spaces in " stripped string are also bad, it did not occurred to me that it can be intentional. Will fix.

Reply:
> Can you explain why you want to strip whitespace that the user included explicitly?

actually, existing code did do a strip, so maybe we should keep that, see also: https://github.com/zephyrproject-rtos/zephyr/pull/30214#discussion_r604750095

Especially `CONFIG_TEST="My string"    ` (<-- notice trailing whitespaces) will fail, as the `"` is no longer matched on `value.endswith('"')`

Reply:
@tejlmand if I understand you, you're suggesting we do this instead:

```py
                    value = match.group('value').rstrip()                              # notice: rstrip(), NOT strip()
                    if value.startswith('"') and value.endswith('"'):
                        value = value[1:-1]
```

Instead of the current `value = match.group('value')`. That would allow _trailing_ whitespace in the value, which is consistent with what `BuildConfiguration._parse` does today, but is not consistent with `load_dot_config()`. We have to pick one behavior, though, and I think allowing trailing whitespace like `BuildConfiguration` does makes sense, especially given how sloppy a lot of modern editors are when it comes to inserting it.

The current code in both `load_dot_config()` and `BuildConfiguration._parse` does _not_ strip _leading_ whitespace from the value though, so `strip()` would not be consistent with either one. That is, `CONFIG_FOO=     y` is not treated the same as `CONFIG_FOO=y` by either function, and I don't think we should make a behavior change here.

Reply:
> 

> @tejlmand if I understand you, you're suggesting we do this instead:

Not necessarily there, but that would be a fine place to do it.
I was merely observing that the regex would include trailing whitespaces, but that didn't happen with old code, that looked like:
```
line = line.strip()
if not line or line.startswith('#'):
    continue
option, value = line.split('=', 1)
```

Of course there can be good reasons to change old behavior, but as you so nicely put it:
> especially given how sloppy a lot of modern editors are when it comes to inserting it.

then I think the `value = match.group('value').rstrip()` proposal looks very nice.
I rarely think users places whitespaces in front of settings, like: `   CONFIG_TEST=y`.
More common to get unintended trailing whitespaces.

Note: whitespaces in front of values has never been stripped, so I don't think we should start doing that.

Reply:
OK, sounds like we are on the same page and the requested change is to use:

```py
                    value = match.group('value').rstrip()
```

Thanks for catching this.